Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -319,26 +319,21 @@ template<typename... Elements> struct ArgumentCoder<std::tuple<Elements...>> {
             (encoder << ... << std::get<Indices>(tuple));
     }
 
-    template<typename Decoder>
-    static std::optional<std::tuple<Elements...>> decode(Decoder& decoder)
-    {
-        return decode(decoder, std::tuple<> { }, std::index_sequence_for<> { });
-    }
-
-    template<typename Decoder, typename OptionalTuple, size_t... Indices>
-    static std::optional<std::tuple<Elements...>> decode(Decoder& decoder, OptionalTuple&& optionalTuple, std::index_sequence<Indices...>)
+    template<typename Decoder, typename... DecodedTypes>
+    static std::optional<std::tuple<Elements...>> decode(Decoder& decoder, std::optional<DecodedTypes>&&... decodedObjects)
     {
-        constexpr size_t Index = sizeof...(Indices);
-        static_assert(Index == std::tuple_size_v<OptionalTuple>);
+        constexpr size_t Index = sizeof...(DecodedTypes);
```

Comment:
I know I missed it the last time, but now that you're touching the lines:
Maybe the constexpr constants could follow normal webkit variable naming. E.g. `index`, `shouldHandleElement`, etc.
shouldHandleElement has extra parenthesis.

Reply:
OK.